packages/transformers/webmanifest/src/dependencies.js

```diff
@@ -0,0 +1,36 @@
+// @flow
+
+import type {MutableAsset} from '@parcel/types';
+
+import nullthrows from 'nullthrows';
+
+const handleSrcItem = (asset, {src}) => {
+  asset.addURLDependency(src, {});
+};
+
+const handleArray = (asset, list) => {
+  if (!Array.isArray(list)) {
+    return;
+  }
+  list.forEach(handleSrcItem.bind(null, asset));
+};
+
+// A list of all properties that may produce a dependency
+// Based on https://developer.mozilla.org/en-US/docs/Web/Manifest#Members
+const HANDLERS = {
+  icons: handleArray,
+  screenshots: handleArray,
+  serviceworker: handleSrcItem
+};
+
+export default function collectDependencies(
+  asset: MutableAsset,
+  json: {[string]: any}
+) {
+  nullthrows(json);
+  Object.keys(HANDLERS).forEach(key => {
+    if (json[key]) {
+      HANDLERS[key](asset, json[key]);
```

Comment:
The url rewriting should happen here: if you return the result of `asset.addURLDependency`, and replace `json[key]` with it, it will be replaced with a URL in a later build step.